lib/route/ROUTING.md

```diff
@@ -0,0 +1,491 @@
+    toText = T.pack . show . unUserId
+```
+
+Next to encode `Text` from the route input as a `UserId`. This means we have to check that this
+input is valid. Our requirement is that the input is a number greater than zero:
+
+```haskell
+userIdTextPrism :: Prism' Text UserId
+userIdTextPrism = prism toText fromText
+  where
+    -- Unwrap the UserId, use 'show' on the 'Int' value, and then turn that into 'Text'
+    toText :: UserId -> Text
+    toText = Text.pack . show . unUserId
+
+    fromText :: Text -> Either Text UserId
+    fromText t = case readMaybe $ Text.unpack x of
+      -- Match when we have successfully read an Int value
+      -- and use a 'guard' to ensure it is greater than zero.
+      Just i | i > 0 -> Right $ UserId i
+      -- For everything else we return a Left value to indicate a failure.
+      _ -> Left x
+```
+
+We pass our `Prism'` to the aptly named `reviewEncoder` function from `Obelisk.Route` to create
+the `Encoder`. In doing so we avoid the boring bits of having to chop up the route ourselves:
+
+```haskell
+reviewEncoder userIdTextPrism :: Encoder parse check UserId Text
+```
+
+Add this to the end of our `case` expression in `Common.Route`:
+
+```haskell
+FrontendRoute_User -> PathSegment "user" $ _todo . reviewEncoder userIdTextPrism
+```
+
+Next is to tell Obelisk that we have only a single path segment and once it has matched the
+static part and the `UserId` this route is complete. This is done by composing our `UserId`
+encoder with the `singlePathSegmentEncoder` using `(.)` imported from `Control.Category`:
+
+#### A more general composition
+
+Importantly the composition operator we need is the more general version from the
+`Control.Category` module. This is because the one from `Prelude` is specialised to the function
+arrow type `(->)`, whereas the one in `Control.Category` uses the `Category` typeclass:
+
+```haskell
+-- (.) from Prelude
+(.) :: (b -> c) -> (a -> b) -> a -> c
+
+-- (.) from Control.Category
+(.) :: Category cat => cat b c -> cat a b -> cat a c
+```
+
+Ensure that `(.)` is imported correctly at the top of `Common.Route`:
+
+ ```haskell
+import Prelude hiding ((.))
+import Control.Category
+```
+
+----
+
+Our addition to the `case` expression should look like this:
+
+```haskell
+  FrontendRoute_User -> PathSegment "user" $ singlePathSegmentEncoder . reviewEncoder userIdTextPrism
+```
+
+It should be a rare event that you will need to write your own `Encoder` function like
+`reviewEncoder` or `singlePathSegmentEncoder` directly. The provided `Encoder`s in `Obelisk.Route`
+are designed to be composed and combined to build more complicated encoders and should be
+sufficient for most requirements. If you find this is not the case and there is a pattern that
+keeps appearing and is not accounted for, then that is probably a bug and we'd like to know about
+it!
+
+#### Aside: 'Wrapped' typeclass and Template Haskell.
```

Comment:
Note: the `Wrapped` class doesn't require TemplateHaskell, if you use `DeriveGeneric` (or a `Generic` instance) the default class implementation will use that.